### PR TITLE
Added onReady, onEnd, onError, and onClose event handlers to config object. 

### PR DESCRIPTION
**Pull request by samhiatt**
Allows user to add connection event handlers, for example, to enable automatic reconnect on connection close event.

### CODE REVIEW

**Comment by agebrock** [comment]
Thx @samhiatt,
your changes will be reviewed soon.

**Comment by samhiatt** [comment]
Cool, thanks. ... BTW, I think the interface could use some work, like I'm not sure if the the config object is the best place for the event handlers, and might be confusing since 'connect' takes a callback already. At any rate, I'm happy to discuss and modify my changes if you have other ideas. :)

File: .gitignore [hunk 1]
```diff
@@ -1,25 +1,2 @@
-# Logs
-logs
-*.log
+.*
```
Comment (agebrock) [inline .gitignore]: Why did you removed the content of this file ?

Comment (samhiatt) [inline .gitignore]: Shoot, sorry, I don't think I did that intentionally.

**Comment by agebrock** [comment]
Hi @samhiatt see https://github.com/Finanzchef24-GmbH/tunnel-ssh/blob/master/example/events.js
for an example how to interact with events.

**Comment by samhiatt** [comment]
Awesome! I noticed that over the weekend and began testing it. Looks like a
much cleaner implementation. Nice work!

**Pull request closed without merging**